packages/server/lib/cloud/protocol.ts

```diff
@@ -173,39 +182,53 @@ export class ProtocolManager implements ProtocolManagerShape {
     this.invokeSync('resetTest', testId)
   }
 
-  async uploadCaptureArtifact ({ uploadUrl, timeout }) {
-    const dbPath = this._dbPath
+  canUpload (): boolean {
+    return !!this._protocol && !!this._dbPath && !!this._db
+  }
 
-    if (!this._protocol || !dbPath || !this._db) {
-      if (this._errors.length) {
-        await this.sendErrors()
-      }
+  hasErrors (): boolean {
+    return !!this._errors.length
+  }
 
+  async getZippedDb (): Promise<Buffer | void> {
+    const dbPath = this._dbPath
+
+    debug('reading db from', dbPath)
+    if (!dbPath) {
```

Comment:
Maybe add a debug statement here so we can follow if this is happening.

Reply:
Given this method is going away soon enough, I'd as soon leave it as is